tests/device_tests/test_firmware_hash.py

```diff
@@ -2,21 +2,22 @@
 
 import pytest
 
-from trezorlib import firmware
+from trezorlib import firmware, models
 from trezorlib.debuglink import TrezorClientDebugLink as Client
 
 FIRMWARE_LENGTHS = {
-    "1": 7 * 128 * 1024 + 64 * 1024,
-    "T": 13 * 128 * 1024,
-    "Safe 3": 13 * 128 * 1024,
+    models.T1B1: 7 * 128 * 1024 + 64 * 1024,
+    models.T2T1: 13 * 128 * 1024,
+    models.T2B1: 13 * 128 * 1024,
+    models.T3T1: 13 * 128 * 1024,
```

Comment:
Not directly related to this PR, but how is this value determined? And can we link some source here?

Reply:
Thinking about it, it should probably correspond to the overall size of `FIRMWARE_AREA`, and thus is actually wrong for T3T1. Should be fixed in 0079dc0045391b8e615caf2d162e447320a0296c